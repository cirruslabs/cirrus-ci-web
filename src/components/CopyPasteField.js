import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import MuiTextField from '@material-ui/core/TextField';

const styles = theme => ({
  input: {
    '& input[readonly]': {
      padding: '10px 14px',
      borderRadius: theme.shape.borderRadius,
      backgroundColor: theme.palette.grey[100],
    },
  },
});

function CopyPasteField(props) {
  const { classes, InputProps = {}, ...other } = props;

  return (
    <MuiTextField
      onFocus={event => {
        event.target.select();
      }}
      InputProps={{
        ...InputProps,
        disableUnderline: true,
        className: classNames(classes.input, InputProps.className),
      }}
      {...other}
    />
  );
}

CopyPasteField.propTypes = {
  classes: PropTypes.object.isRequired,
  InputProps: PropTypes.object,
};

export default withStyles(styles)(CopyPasteField);
