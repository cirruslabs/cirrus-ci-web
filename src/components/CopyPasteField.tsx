import React from 'react';
import classNames from 'classnames';
import { withStyles, createStyles, WithStyles } from '@material-ui/core/styles';
import MuiTextField, { StandardTextFieldProps } from '@material-ui/core/TextField';

const styles = theme =>
  createStyles({
    input: {
      '& input[readonly]': {
        padding: '10px 14px',
        borderRadius: theme.shape.borderRadius,
        backgroundColor: theme.palette.grey[100],
      },
    },
  });

type Props = WithStyles<typeof styles> & StandardTextFieldProps;

function CopyPasteField(props: Props) {
  const { classes, InputProps = {}, ...other } = props;

  return (
    <MuiTextField
      onFocus={event => {
        event.target.select();
        document.execCommand('copy');
      }}
      InputProps={{
        ...InputProps,
        disableUnderline: true,
        className: classNames(classes.input, InputProps.className),
      }}
      {...other}
    />
  );
}

export default withStyles(styles)(CopyPasteField);
