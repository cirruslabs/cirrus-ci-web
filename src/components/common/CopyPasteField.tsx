import React from 'react';
import { IconButton, InputAdornment, OutlinedInput } from '@mui/material';
import CopyIcon from '@mui/icons-material/FileCopy';
import copy from 'clipboard-copy';
import { OutlinedInputProps } from '@mui/material/OutlinedInput/OutlinedInput';

function CopyPasteField(props: OutlinedInputProps) {
  return (
    <OutlinedInput
      value={props.value}
      contentEditable={false}
      endAdornment={
        <InputAdornment position="end">
          <IconButton
            aria-label="copy"
            onClick={event => {
              copy(props.value!.toString());
            }}
            edge="end"
            size="large"
          >
            <CopyIcon />
          </IconButton>
        </InputAdornment>
      }
      {...props}
    />
  );
}

export default CopyPasteField;
