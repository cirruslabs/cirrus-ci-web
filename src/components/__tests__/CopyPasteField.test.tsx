import React from 'react';
import CopyPasteField from '../common/CopyPasteField';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<CopyPasteField name="my copy-paste field" value={'hello world'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
