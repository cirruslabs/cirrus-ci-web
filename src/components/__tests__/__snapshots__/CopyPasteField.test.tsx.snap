// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root CopyPasteField-input-1 MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      name="my copy-paste field"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="hello world"
    />
  </div>
</div>
`;
